SalesManagement/SalesManagement/OnlineShop.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace SalesManagement
+{
+    public class OnlineShop
+    {
+        public event EventHandler<GoodsInfoEventArgs> OnNewGoods;
+
+        public void NewGoods(string goodsName)
+        {
+            if (this.OnNewGoods == null)
+                throw new InvalidOperationException("Event OnNewGoods must have subscribers.");
```

Comment:
I think throwing exception here is unnecessary. No customers, no messages, but shop have to work. But it just my vision, business logic can be different)